src/coreclr/jit/gentree.h

```diff
@@ -7661,7 +7661,7 @@ inline bool GenTree::IsSIMDZero() const
 //
 inline bool GenTree::IsFloatPositiveZero() const
 {
-    return !(IsCnsNonZeroFltOrDbl());
+    return IsCnsFltOrDbl() && !IsCnsNonZeroFltOrDbl();
```

Comment:
I wonder if we just should explicitly implement the logic?
```c++
    if (IsCnsFltOrDbl())
    {
        double constValue = AsDblCon()->gtDconVal;
        return *(__int64*)&constValue == 0;
    }
```

For me personally, a conjunction of *"is a floating point constant?"* and *"is not a non-zero floating point constant"* is tricky to parse mentally.

Reply:
If you agree you can follow up after the snap - don't want to block the PR on that.

Reply:
Yea, we can follow up and I agree that the current logic is hard to parse mentally which is probably why there was a bug :)